protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlOperationMetadata.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.protocols.xml.internal.unmarshall;
+
+import software.amazon.awssdk.annotations.NotThreadSafe;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+
+/**
+ * Contains information needed to create a {@link AwsXmlResponseHandler} for the client.
+ */
+@NotThreadSafe
+@SdkProtectedApi
+public final class XmlOperationMetadata {
+
+    private boolean hasStreamingSuccessResponse;
+    private boolean useRootElement;
+
+    public boolean isHasStreamingSuccessResponse() {
+        return hasStreamingSuccessResponse;
+    }
+
+    public XmlOperationMetadata withHasStreamingSuccessResponse(boolean hasStreamingSuccessResponse) {
+        this.hasStreamingSuccessResponse = hasStreamingSuccessResponse;
+        return this;
+    }
+
+    public boolean useRootElement() {
+        return useRootElement;
+    }
+
+    /**
+     * This value indicates if the root element in the response xml document should be ignored
```

Comment:
Do we need this or can we just use the PayloadTrait?

Reply:
As we discussed yesterday, we are setting a customization to enable this value for operation like S3 `GetBucketLocationOutput`. So we need it

Reply:
I was thinking that we would use that customization to just apply the payload trait to LocationConstraint. Wouldn't that have the same behavior?